qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -568,6 +571,74 @@ def __repr__(self) -> str:
         )
 
 
+class ScalableSymbolicPulse(SymbolicPulse):
+    r"""Subclass of :class:`SymbolicPulse` for pulses with scalable envelope.
+
+    Instance of :class:`ScalableSymbolicPulse` behaves the same as an instance of
+    :class:`SymbolicPulse`, but its envelope is assumed to have a scalable form
+    :math:`\text{amp}\times\exp\left(i\times\text{angle}\right)\times\text{F}
+    \left(t,\text{parameters}\right)`,
+    where :math:`\text{F}` is some function describing the rest of the envelope.
+    Parameters `amp` and `angle` are assumed to appear in :attr:`SymbolicPulse.parameters`.
+
+    When two :class:`ScalableSymbolicPulse` objects are equated, instead of comparing
+    `amp` and `angle` individually, only the complex amplitude
+    :math:'\text{amp}\times\exp\left(i\times\text{angle}\right)' is compared.
+    """
+
+    def __init__(
+        self,
+        pulse_type: str,
+        duration: Union[ParameterExpression, int],
```

Comment:
Since you assume amp and angle exist in the equality evaluation, I think it's more straightforward to add `amp` and `angle` (and do typecheck) to constructor and update parameters dictionary to guarantee these parameters exist.

Reply:
I thought about this too, but wasn't sure what's the right way to go.
If you think it's better - I'll move it over to the class.